lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -1524,3 +1543,51 @@ def offload_agw(self, imsi, enbID, offloadtype=0):
             self._ha_stub.StartAgwOffload(req)
         except grpc.RpcError as e:
             print("gRPC failed with %s: %s" % (e.code(), e.details()))
+
+
+class HeaderEnrichmentUtils:
+    def __init__(self):
+        self.magma_utils = MagmadUtil(None)
+        self.dump = None
+
+    def restart_envoy_service(self):
+        print("restarting envoy")
+        self.magma_utils.exec_command_output("sudo service magma@envoy_controller restart")
+        time.sleep(5)
+        self.magma_utils.exec_command_output("sudo service magma_dp@envoy restart")
+        time.sleep(20)
+        print("restarting envoy done")
+
+    def get_envoy_config(self):
+        output = self.magma_utils.exec_command_output(
+            "sudo ip netns exec envoy_ns1 curl 127.0.0.1:9000/config_dump")
+        self.dump = json.loads(output)
+
+        return self.dump
+
+    def get_route_config(self):
+        self.dump = self.get_envoy_config()
+
+        for conf in self.dump['configs']:
+            if 'dynamic_listeners' in conf:
+                return conf['dynamic_listeners'][0]['active_state']['listener']['filter_chains'][0]['filters']
```

Comment:
can we confirm that those nested map and list always will be non-empty? this could throw `AttributeError` if the key or value doesn't exist